src/schema/schema_neo4j_queries.py

```diff
@@ -112,6 +112,37 @@ def get_entity(neo4j_driver, uuid):
     return result
 
 
+
+"""
+Get the uuids for each entity in a list that doesn't belong to a certain entity type. Uuids are ordered by type
+
+Parameters
+----------
+neo4j_driver : neo4j.Driver object
+    The neo4j database connection pool
+direct_ancestor_uuids : list
+    List of the uuids to be filtered
+
+Returns
+-------
+dict
+    A dictionary of entity uuids that don't pass the filter, grouped by entity_type
+"""
+def filter_ancestors_by_type(neo4j_driver, direct_ancestor_uuids, entity_type):
+    uuids_str = ", ".join([f'"{uuid}"' for uuid in direct_ancestor_uuids])
+    query = (f"MATCH (e:Entity) "
+             f"WHERE e.uuid in [{uuids_str}] AND toLower(e.entity_type) <> '{entity_type.lower()}' "
```

Comment:
You can use `WHERE e.uuid in {direct_ancestor_uuids}` directly without joining the uuids into `uuids_str`, can be more efficient.